extensions/ql-vscode/src/model-editor/modeled-method.ts

```diff
@@ -126,3 +126,24 @@ export function isModelAccepted(
     modeledMethod.provenance !== "ai-generated"
   );
 }
+
+export function calculateNewProvenance(
+  modeledMethod: ModeledMethod | undefined,
+) {
+  if (!modeledMethod || !modeledMethodSupportsProvenance(modeledMethod)) {
+    return "manual";
+  }
+
+  switch (modeledMethod.provenance) {
```

Comment:
Does it make sense to add a comment here? I think it can be rather confusing if the issue is not read carefully parallel to this code.

Reply:
Good point. I've added some comments, please let me know if you have any suggestions to make things better.